tdigest.go

```diff
@@ -225,15 +225,13 @@ func (t *TDigest) Compress() (err error) {
 	}
 
 	oldTree := t.summary
-	t.summary = newSummary(t.summary.Len())
+	t.summary = newSummary(estimateCapacity(t.compression))
```

Comment:
Please check this change in Compress. Previously we created summary of the same length which does not make much sense...

Reply:
Nice catch! The old code would set it up to grow the summary as soon as a summary.Add triggered :x